tests/jballerina-unit-test/src/test/resources/test-src/record/readonly_record_fields.bal

```diff
@@ -171,6 +171,19 @@ type Identifier record {|
     string code;
 |};
 
+type Foo2 record {|
+    function () f;
+    int id;    
+|};
+
+function testRecordWithFunctionTypeField() {
+    Foo2 & readonly x = {
+        f: testRecordWithFunctionTypeField,
```

Comment:
Let's add some tests where we call these functions too. Let's use different combinations of param and return types.